### PR TITLE
Feature/uar 616 oe details add transaction on url

### PR DESCRIPTION
**Pull request by AndreKramer7654**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-616

### Change description
Add transaction id and set up interceptor config
Re-enable partial validation for entity details

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesDataController.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.TRANSACTION_KEY;
 
 @RestController
-@RequestMapping("/overseas-entity-details")
+@RequestMapping("/private/transactions/{transaction_id}/overseas-entity/{overseas_entity_id}/details")
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesDataController.java:30]: follow pattern for "filings" endpoint.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java [hunk 2]
```diff
@@ -133,7 +133,7 @@ private void validateTrustDetails(OverseasEntitySubmissionDto overseasEntitySubm
 
     public Errors validatePartial(OverseasEntitySubmissionDto overseasEntitySubmissionDto, Errors errors, String loggingContext) {
         if (isRoeUpdateEnabled && overseasEntitySubmissionDto.isForUpdate()) {
-            // validatePartialUpdateDetails(overseasEntitySubmissionDto, errors, loggingContext);
+            validatePartialUpdateDetails(overseasEntitySubmissionDto, errors, loggingContext);
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:136]: This re-enables the validation taken out for uar-616 failures in reserve1/cidev.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java [hunk 3]
```diff
@@ -133,7 +133,7 @@ private void validateTrustDetails(OverseasEntitySubmissionDto overseasEntitySubm
 
     public Errors validatePartial(OverseasEntitySubmissionDto overseasEntitySubmissionDto, Errors errors, String loggingContext) {
         if (isRoeUpdateEnabled && overseasEntitySubmissionDto.isForUpdate()) {
-            validatePartialUpdateDetails(overseasEntitySubmissionDto, errors, loggingContext);
+            // validatePartialUpdateDetails(overseasEntitySubmissionDto, errors, loggingContext);
```
Comment (bantao-ch) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java]: Why comment out here ?

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java]: This was commented out to fix breaking change and can't be enabled. There's a ticket to fix the validation UAR-711.

**Review by bantao-ch — APPROVED** [review]

**Review by jkayyala — APPROVED** [review]

**Review by bmcgreevyCH — APPROVED** [review]

**Review by stephanie-ch — APPROVED** [review]

**Pull request merged**